### ISSUE TITLE
CLI Docs

### ISSUE THREAD

**Issue by Akshay-Hegde** [photoview/photoview#818]
Is there any CLI to manage Albums ?
Example : List Albums, Share Albums etc

Also is there any possibility to share all albums to public, may be a root directory ?

Any way to automate scanning process using crontab etc, like Original Directory is changed or Renamed, for re-indexing album

**Comment by WindLi001** [comment]
> Is there any CLI to manage Albums ?
No. There is NO CLI in Photoview at all.

> Also is there any possibility to share all albums to public, may be a root directory ?
Maybe no.

> Any way to automate scanning process using crontab etc, like Original Directory is changed or Renamed, for re-indexing album
Your can set the auto scan interval in Photoview itself, without using crontab.

**Comment by Akshay-Hegde** [comment]
@WindLi001 : I renamed directory, also kept interval of 5 Minutes its already been 8 Hours, Album name not changed.

Any API documentation available ? I just wanted to share every album

**Comment by WindLi001** [comment]
I use interval of 1 day, and in fact I never test the effective interval. But my album will refreshed at some time.
BTW, there is NO API documentation. In fact the only API of Photoview is the GraphQL API provided by the golang backend, but there is no document of it. You have to read the code.

**Comment by viktorstrate** [comment]
> BTW, there is NO API documentation. In fact the only API of Photoview is the GraphQL API provided by the golang backend, but there is no document of it. You have to read the code.

GraphQL is self-documenting. The entire API is documented by the `schema.graphql` file.[^schema]
I have put a lot of effort in commenting everything in the file to make it as understandable as possible.
Please ask if you have further questions regarding the API.

[^schema]: https://github.com/photoview/photoview/blob/master/api/graphql/schema.graphql

**Comment by viktorstrate** [comment]
This is not supported (see #624 for further discussion on this).

I think it will be hard to make work just by adding headers to the requests, since the client (browser) holds its own state of which it is authenticated or not.

**Comment by kkovaletp** [comment]
@jordy2254, I'd like to close this issue as not planned: we're not going to provide any CLI in the scope of this project soon.

However, in the comments, there is a discussion about the need to have the possibility of sharing all media (root album) with everyone. I remember that we've discussed a user role for guest users, but I don't remember the ticket ID for this. As you are working on this epic, could you please double-check that the mentioned use case is covered in the requirements? Probably it also makes sense to open a new issue with this feature request - feel free to decide).

Let me know once it is done, so I can close this ticket.

**Comment by jordy2254** [comment]
@kkovaletp I agree the cli is out of scope and as per viktor's response everything available in the ui is available via the graphql api which is well documented. It's possible to write a cli but out of scope of the project at this moment in time and many use cases I can think of (including ops) would be better suited as a one time script integrating with the instance.

@Akshay-Hegde Thanks for raising the issue, and if you have further use cases in the future or want to discuss more in depth feel free to open another issue/reopen this one.
